$OpenBSD: patch-vncstorepw_ultravnc_dsm_helper_c,v 1.1 2015/09/16 10:29:43 sthen Exp $

LibreSSL 2.3 gets rid of SHA-0 (EVP_sha). There is no suitable OPENSSL_NO_xxx.

--- vncstorepw/ultravnc_dsm_helper.c.orig	Thu Apr 22 22:29:43 2010
+++ vncstorepw/ultravnc_dsm_helper.c	Wed Sep 16 04:27:01 2015
@@ -413,8 +413,10 @@ void enc_do(char *ciph, char *keyfile, char *lport, ch
 		p++;
 		if (strstr(p, "md5+") == p) {
 			Digest = EVP_md5();        p += strlen("md5+");
+#if !(LIBRESSL_VERSION_NUMBER >= 0x20030000L)
 		} else if (strstr(p, "sha+") == p) {
 			Digest = EVP_sha();        p += strlen("sha+");
+#endif
 		} else if (strstr(p, "sha1+") == p) {
 			Digest = EVP_sha1();       p += strlen("sha1+");
 		} else if (strstr(p, "ripe+") == p) {
